ext/readline/readline_cli.c

```diff
@@ -617,7 +617,12 @@ static int readline_shell_run(void) /* {{{ */
 	}
 
 #ifndef PHP_WIN32
-	history_file = tilde_expand("~/.php_history");
+#define PHP_HISTFILE_ENV "PHP_HISTFILE"
+	if (getenv(PHP_HISTFILE_ENV)) {
+		spprintf(&history_file, MAXPATHLEN, "%s", getenv(PHP_HISTFILE_ENV));
+	} else {
+		spprintf(&history_file, MAXPATHLEN, "%s/.php_history", getenv("HOME"));
```

Comment:
This path is no longer resolved, but I can't find any reference in `read_history` or `append_history` that it must be. So I suppose this is fine.